docs/document/docs/guides/attachment-customization.md

```diff
@@ -0,0 +1,176 @@
+---
+id: attachment-customization
+title: Attachment customization
+sidebar_label: Attachment customization
+---
+
+## Overview
+This article assumes the following scenario: 
+1. Create a file object (blob/arrayBuffer pattern and simple object pattern) and apply it to KUC Attachment component
+2. Get the file info that user attached
+3. Validate the file type/size and show error in case it is invalid
+4. Apply KUC Attachment component files into native kintone Attachment field
+
+### Components to use
+- [Attachment](../components/desktop/attachment.md)
+- [Button](../components/desktop/button.md)
+- [Spinner](../components/desktop/spinner.md)
+
+# Completed image
+
+The completed image of the customized page is as follows:
+![attachment customize](assets/attachment_customize.gif)
+
+## JavaScript and CSS Customization
+
+When you import the UMD file of Kintone UI Component to the app, you can upload the JavaScript files by following these steps:
+
+You can see how to upload a file in the [Quick Start](../getting-started/quick-start.md).
+
+
+## What you will need to have ready
+
+Create a app that include a attachment field with id "Attachment" and a blank space filed with id "space". 
+
+### Display custom attachment area
+Display the kuc attachment component(We can directly add the files in the native kintone attachment field to the files property of kuc attachment) and two buttons:
+- add custom files to KUC Attachment
+- upload to native kintone Attachment
+
+```javascript
+kintone.events.on('app.record.detail.show',async (event)=>{
+  if(event.record[`${ATTACHMENT_FIELD_ID}`]){
+    const attachment = new Kuc.Attachment({files: record[`${ATTACHMENT_FIELD_ID}`].value, label:'KUC Attachment'});
+    const addCustomFilesButton = new Kuc.Button({text: 'add custom files to KUC Attachment'});
+    const uploadButton = new Kuc.Button({text: 'upload to native kintone Attachment'});
+    const spinner = new Kuc.Spinner();
+    const spaceElement = kintone.app.record.getSpaceElement(SPACE_ID);
+    const container = document.createElement('div');
+    container.appendChild(attachment);
+    container.appendChild(addCustomFilesButton);
+    container.appendChild(uploadButton);
+    spaceElement.appendChild(container);
+  }
+  return event;
+});
+```
+### Create some file objects and apply them to KUC Attachment component
+Add a click event listener for addCustomFilesButton. 
+When the button was clicked it will create three types of file objects(modify blob/arrayBuffer to [File object](https://developer.mozilla.org/en-US/docs/Web/API/File) and simple object pattern{name: "xx", size: "xx"})
+and add them to the file properties of the attachment.
```

Comment:
```suggestion
Add a click event listener for `addCustomFilesButton`.<br>
When the button is clicked, it will create three types of file objects as follows:
- Blob/ArrayBuffer files modified to [File object](https://developer.mozilla.org/en-US/docs/Web/API/File)
- Simple object (`{name: "xx", size: "xx"}`)
And add them to the `files` property of the KUC Attachment component.
```